### PR TITLE
adding system test and circleci config

### PR DESCRIPTION
**Pull request by esc**

### CODE REVIEW

File: system_test.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+""" This is the system test for texasbbq.
+
+It tests that the latest released version of Numba with the latest released
+version of umap. If anything is red, it's probably due to a bug with texasbbq.
```
Comment (stuartarchibald) [inline system_test.py]:
```suggestion
version of umap. If anything fails, it's probably due to a bug in texasbbq.
```

Comment (stuartarchibald) [inline system_test.py]: I presume `umap` is chosen for its stability and use of just Numba's jit decorators?

Comment (esc) [reply system_test.py]: Indeed, it's most likely to always be successful.

Comment (esc) [reply system_test.py]: Thanks, being a bit too colloquial here.

**Pull request merged**